src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPMappingService.kt

```diff
@@ -38,4 +44,21 @@ class CSIPMappingService(@Qualifier("mappingApiWebClient") webClient: WebClient)
       .retrieve()
       .awaitBodilessEntity()
   }
+
+  suspend fun createCSIPFactorMapping(
+    mapping: CSIPFactorMappingDto,
+  ): CreateMappingResult<CSIPFactorMappingDto> {
+    return webClient.post()
+      .uri("/mapping/csip/factors")
+      .bodyValue(
+        mapping,
+      )
+      .retrieve()
+      .bodyToMono(Unit::class.java)
+      .map { CreateMappingResult<CSIPFactorMappingDto>() }
```

Comment:
I don't really understand this. You get an empty response and then create a new result? Wouldn't you just get the result returned from the API call?

Reply:
yes - that needs tidying up - will update in next PR

Reply:
Had another look at this and it's because I have copied the pattern from other child mappings which replicate the code in MappingApiService.  Saying that, I'll see if it can be tidied up.